pymc/sampling/jax.py

```diff
@@ -144,13 +144,15 @@ def get_jaxified_graph(
     return jax_funcify(fgraph)
 
 
-def get_jaxified_logp(model: Model, negative_logp=True) -> Callable:
+def get_jaxified_logp(
+    model: Model, negative_logp=True
+) -> Callable[[Sequence[np.ndarray]], np.ndarray]:
     model_logp = model.logp()
     if not negative_logp:
         model_logp = -model_logp
     logp_fn = get_jaxified_graph(inputs=model.value_vars, outputs=[model_logp])
 
-    def logp_fn_wrap(x):
+    def logp_fn_wrap(x: Sequence[np.ndarray]) -> np.ndarray:
```

Comment:
This is not correct, it takes jax arrays and outputs jax arrays

Reply:
I don't think that's 100% true. Checking with the interactive debugger confirms that the return type is `jax.Array`, but the initial point functions return a `dict[str, np.ndarray]`, and we can successfully pass the `.values()` of that dict into the jaxified function. So it can seemingly accept anything that's coercible to an array. Maybe it's more correct to annotate it like this:

``` python
def logp_fn_wrap(x: Sequence[ArrayLike]) -> jax.Array:
```
ArrayLike is from `numpy.typing`: https://numpy.org/devdocs/reference/typing.html#numpy.typing.ArrayLike

Reply:
I've just pushed a commit to improve this, it's a bit tricky to annotate at the interface with `_init_jitter` given that `jax` is an optional dependency. I've left the type annotation as returning a `np.ndarray` but included that it may return a `jax.Array` in the docstring.